### ISSUE TITLE
Avoid direct calls to session_*() functions

### ISSUE THREAD

**Issue by elazar** [trvshrmn/slim-skeleton#6]
Consider using a library like [`illuminate/session`](https://github.com/illuminate/session) or [`aura/session`](https://github.com/auraphp/Aura.Session) instead.

**Comment by trvshrmn** [comment]
I looked at those libraries. Looks like it will take some work to get either one going. I like Laravel and Illuminate. Will the illuminate library require that I add a sessions table in the database?

Sessions are something I want to get right. At the moment I am using session_*() functions or $_SESSION on about 5-6 lines. Can you help me understand the need for whole new library? What are some vulnerabilities or downsides to doing it the way I have been?

**Comment by elazar** [comment]
> Will the illuminate library require that I add a sessions table in the database?

Not necessarily. It has [several drivers](https://laravel.com/docs/5.3/session#configuration) you can choose from; not all of them require a database.

> At the moment I am using session_*() functions or $_SESSION on about 5-6 lines. Can you help me understand the need for whole new library? What are some vulnerabilities or downsides to doing it the way I have been?

Using [superglobals](http://php.net/manual/en/language.variables.superglobals.php) directly is generally not a good practice for the same reason that [global variables](http://wiki.c2.com/?GlobalVariablesAreBad) should generally be avoided, in particular because changes to them are difficult to trace and can make debugging and automated testing difficult.

Some libraries can also help to mitigate potential issues like [session fixation](https://en.wikipedia.org/wiki/Session_fixation) and [session poisoning](https://en.wikipedia.org/wiki/Session_poisoning) vulnerabilities.
